crates/oxide/src/lib.rs

```diff
@@ -427,35 +451,59 @@ impl Scanner {
 }
 
 fn read_changed_content(c: ChangedContent) -> Option<Vec<u8>> {
-    if let Some(content) = c.content {
-        return Some(content.into_bytes());
-    }
-
-    let Some(file) = c.file else {
-        return Default::default();
-    };
+    let (content, extension) = match c {
+        ChangedContent::File(file, extension) => match std::fs::read(&file) {
+            Ok(content) => (content, extension),
+            Err(e) => {
+                event!(tracing::Level::ERROR, "Failed to read file: {:?}", e);
+                return None;
+            }
+        },
 
-    let Ok(content) = std::fs::read(&file).map_err(|e| {
-        event!(tracing::Level::ERROR, "Failed to read file: {:?}", e);
-        e
-    }) else {
-        return Default::default();
+        ChangedContent::Content(contents, extension) => (contents.into_bytes(), extension),
```

Comment:
Refactored this such that we can have access to the content _and_ the extension. 
Before this we weren't applying the preprocessing if the `ChangedContent` was just a `Content` instead of a file.

Now both versions will be preprocessed.